src/impl/tpls/KokkosBlas3_gemm_tpl_spec_decl.hpp

```diff
@@ -452,11 +585,23 @@ struct GEMM< \
       transb = CUBLAS_OP_T; \
     else \
       transb = CUBLAS_OP_C; \
-    KokkosBlas::Impl::CudaBlasSingleton & s = KokkosBlas::Impl::CudaBlasSingleton::singleton(); \
-    if(!A_is_lr && !B_is_lr && !C_is_lr ) \
-      cublasSgemm(s.handle, transa, transb, M, N, K, &alpha, A.data(), LDA, B.data(), LDB, &beta, C.data(), LDC); \
-    if(A_is_lr && B_is_lr && C_is_lr ) \
-      cublasSgemm(s.handle, transb, transa, N, M, K, &alpha, B.data(), LDB, A.data(), LDA, &beta, C.data(), LDC); \
+    \
+    bool useDotBasedGemm = false; \
+    if(   (!A_is_lr && A_t & transb == CUBLAS_OP_N && M*N < 1600) \
```

Comment:
It would be better to name these magic numbers.  For example:
```c++
constexpr int dot_based_gemm_data_size_threshold = 1600;
```

Reply:
Done.